test/validate_test/ip_address_test.exs

```diff
@@ -0,0 +1,22 @@
+defmodule Fields.ValidateIpAddressTest do
+  use ExUnit.Case
+  use ExUnitProperties
+
+  alias Fields.Validate
+
+  test "ipv6 address is valid" do
+    ip_address = "2001:cdba:0000:0000:0000:0000:3257:9652"
```

Comment:
it could be worth finding an `IPv6` generator and running a property based test. 💡 
But given that you sourced the RegEx from SO, it should be fine. 💭